packages/store/operators/src/compose.ts

```diff
@@ -1,6 +1,7 @@
 import { StateOperator } from '@ngxs/store';
+import { NoInfer } from './utils';
 
-export function compose<T>(...operators: StateOperator<T>[]): StateOperator<T> {
+export function compose<T>(...operators: NoInfer<StateOperator<T>[]>): StateOperator<T> {
```

Comment:
sometimes, Typescript was happier about the inferred internals of the function when wrapping `NoInfer` around the entire parameter type. other times, it was happier only wrapping `T`. so I have implemented a mixture with the principle being whatever results in the smallest code change for correct inference.